### PR TITLE
Ci/coverity integration

### PR DESCRIPTION
**Pull request by nateglims**
*Description of changes:* Changes to enable static analysis in CI:

1) Add MISRA config for coverity from `v4_beta` branch. This provides a set of MISRA rules that are unconditionally ignored.
2) Explicitly link `pthreads` and `CMAKE_DL_LIBS`. These are used by SSL and on some environments (such as our build nodes) CMake has trouble reasoning about the existence of these libraries or if they are required. The `CMAKE_THREAD_PREFER_PTHREAD` setting and `${CMAKE_DL_LIBS}` appear to be best practices.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,9 @@ set_property( GLOBAL PROPERTY USE_FOLDERS ON )
 # Use C90.
 set( CMAKE_C_STANDARD 90 )
 set( CMAKE_C_STANDARD_REQUIRED ON )
+set( CMAKE_THREAD_PREFER_PTHREAD ON )
+
+find_package(Threads REQUIRED)
```
Comment (yourslab) [inline CMakeLists.txt:15]: Does it still work if we move this to `demos/mqtt/mqtt_demo_basic_tls/CMakeLists.txt` or `demos/CMakeLists.txt` (preferably the first).

File: demos/mqtt/mqtt_demo_basic_tls/CMakeLists.txt [hunk 2]
```diff
@@ -17,6 +17,8 @@ target_link_libraries(
         mqtt
         OpenSSL::Crypto
         OpenSSL::SSL
+        Threads::Threads
```
Comment (leegeth) [inline demos/mqtt/mqtt_demo_basic_tls/CMakeLists.txt:20]: can you please add some comments here to indicate why Threads need to be linked in here even though we are not directly using Threads in the TLS demo?

Comment (nateglims) [reply demos/mqtt/mqtt_demo_basic_tls/CMakeLists.txt:20]: Sure, but so you're aware CMake will pull in pthreads when linking OpenSSL either way. I just had to make it explicit for the build environment.

File: CMakeLists.txt (hunk 1 above)
Comment (nateglims) [reply CMakeLists.txt]: Appears so.

File: demos/mqtt/mqtt_demo_basic_tls/CMakeLists.txt (hunk 2 above)
Comment (nateglims) [reply demos/mqtt/mqtt_demo_basic_tls/CMakeLists.txt:24]: Added for Thread and the Dynamic Loading library.

**Pull request merged**